test/common/network/connection_impl_test.cc

```diff
@@ -1122,6 +1127,33 @@ TEST_F(MockTransportConnectionImplTest, WriteEndStreamStopIteration) {
   connection_->write(buffer, true);
 }
 
+// Validate that when the transport signals ConnectionEvent::Connected, that we
+// check for pending write buffer content.
+TEST_F(MockTransportConnectionImplTest, WriteReadyOnConnected) {
+  InSequence s;
+
+  // Queue up some data in write buffer, simulating what happens in SSL handshake.
+  const std::string val("some data");
+  Buffer::OwnedImpl buffer(val);
+  EXPECT_CALL(*file_event_, activate(Event::FileReadyType::Write)).WillOnce(Invoke(file_ready_cb_));
+  EXPECT_CALL(*transport_socket_, doWrite(BufferStringEqual(val), false))
+      .WillOnce(Return(IoResult{PostIoAction::KeepOpen, 100, false}));
```

Comment:
I think you should return 0 instead of 100; otherwise you're claiming that 100 bytes were written, but this test scenario is saying that it cannot be written yet (and 100 bytes is more than the size of the write, so it's weird).

Reply:
Please return 0 on this one